js/models/stock.js

```diff
@@ -2,11 +2,24 @@
 function Stock(symbol) {
   this.symbol = symbol;
   this.numShares = 0;
-
-  // your code to search the API and return a Stock object here
-
+  $.ajax({
+      url: "http://dev.markitondemand.com/api/v2/quote/jsonp?symbol=" + this.symbol,
+      type: "get",
+      dataType: "jsonp",
+      context: this
+    }).done(function(response){
+      document.querySelector("#stock-price").innerHTML = response.LastPrice;
+      this.price = response.LastPrice;
+      document.querySelector("#stock-name").innerHTML = response.Name;
+      this.companyName = response.Name;
+    })
 }
 
 Stock.prototype = {
-  totalValue: function() { return this.price * this.numShares; }
+  getPrice: function() { return this.price; },
```

Comment:
You don't actually need a getter function if you've specified `this.price` somewhere. That makes a property public.